src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -367,50 +366,54 @@
+ * at which the pointer is stored. The access operation returns a zero-length native memory segment, backed by a region
+ * of memory whose starting address is the 64-bit value read at the specified offset.
  * <p>
- * First, clients can unsafely resize a zero-length memory segment by {@linkplain #reinterpret(long) obtaining} a
- * memory segment with the same base address as the zero-length memory segment, but with the desired size,
- * so that the resulting segment can then be accessed directly, as follows:
+ * The returned zero-length memory segment cannot be accessed directly by the client: since the size of the segment
+ * is zero, any access operation would result in out-of-bounds access. Instead, the client must, <em>unsafely</em>,
+ * assign new spatial bounds to the zero-length memory segment. This can be done via the
+ * {@link #reinterpret(long)} method, as follows:
  *
  * {@snippet lang = java:
- * MemorySegment foreign = someSegment.get(ValueLayout.ADDRESS, 0); // size = 0
- *                                    .reinterpret(4)               // size = 4
- * int x = foreign.get(ValueLayout.JAVA_INT, 0);                    // ok
+ * MemorySegment z   =    segment.get(ValueLayout.ADDRESS, ...);   // size = 0
+ * MemorySegment ptr =    z.reinterpret(16);                       // size = 16
+ * int x = ptr.getAtIndex(ValueLayout.JAVA_INT, 3);                // ok
  *}
  * <p>
- * In some cases, a client might additionally want to assign new temporal bounds to a zero-length memory segment.
- * This can be done using the {@link #reinterpret(long, Arena, Consumer)} method, which returns a
- * new native segment with the desired size and the same temporal bounds as those in the provided arena:
+ * In some cases, the client might additionally want to assign new temporal bounds to a zero-length memory segment.
+ * This can be done via the {@link #reinterpret(long, Arena, Consumer)} method, which returns a
+ * new native segment with the desired size and the same temporal bounds as those of the provided arena:
  *
  * {@snippet lang = java:
- * MemorySegment foreign = null;
+ * MemorySegment pointer = null;
  * try (Arena arena = Arena.ofConfined()) {
- *       foreign = someSegment.get(ValueLayout.ADDRESS, 0)           // size = 0, scope = always alive
- *                            .reinterpret(4, arena, null);          // size = 4, scope = arena.scope()
- *       int x = foreign.get(ValueLayout.JAVA_INT, 0);               // ok
+ *       MemorySegment z   = segment.get(ValueLayout.ADDRESS, ...);  // size = 0, scope = always alive
+ *       MemorySegment ptr = z.reinterpret(16, arena, null);         // size = 4, scope = arena.scope()
+ *       int x = ptr.getAtIndex(ValueLayout.JAVA_INT, 3);            // ok
  * }
- * int x = foreign.get(ValueLayout.JAVA_INT, 0); // throws IllegalStateException
+ * int x = pointer.get(ValueLayout.JAVA_INT, 3);                     // throws IllegalStateException
```

Comment:
This doesn't look right. `pointer` is still `null` at the end. I guess `MemorySegment ptr` should just be `pointer` here?